core/src/main/java/org/infinispan/container/impl/EntryFactoryImpl.java

```diff
@@ -62,34 +66,108 @@ public void init() {
    }
 
    @Override
-   public final void wrapEntryForReading(InvocationContext ctx, Object key, int segment, boolean isOwner) {
+   public final CompletionStage<Void> wrapEntryForReading(InvocationContext ctx, Object key, int segment, boolean isOwner) {
       if (!isOwner && !isL1Enabled) {
-         return;
+         return CompletableFutures.completedNull();
       }
+      CompletionStage<Void> returnedStage = CompletableFutures.completedNull();
```

Comment:
nitpick: can be declared inside the `if (isOwner || readEntry.isL1Entry())`

Reply:
Hrmm, I still need the value during the return in many branches.

Reply:
crap, wrong line. I mean the `expiredStage` (line 77)

Reply:
👍 fixed